### PR TITLE
Feature/reinvent

### PR DESCRIPTION
**Pull request by JaktensTid**

### CODE REVIEW

File: microservices/reinvent/reinvent/api.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+from reinvent.services import FineTuning, Inference
+
+app = FastAPI(
+    title='Reinvent4 API',
+    version='0.1'
+)
+
+ft_router = APIRouter(
+    prefix='/api/v1/fine-tuning',
+    tags=['folding']
```
Comment (timurishmuratov7) [inline microservices/reinvent/reinvent/api.py:19]: folding?

File: microservices/reinvent/reinvent/api_models.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import datetime
+from typing import List
+
+import pydantic
+
+
+
+
+@pydantic.dataclasses.dataclass
+class RunFineTuningJobRequest:
```
Comment (timurishmuratov7) [inline microservices/reinvent/reinvent/api_models.py:12]: So if this is a finetuning job request as far as I understand the protein should be passed on which we finetune, but we pass it separately, should we add it to the request instead as well?

File: microservices/reinvent/reinvent/loggers.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+_logger.setLevel(logging.DEBUG)
+
+_logHandler = logging.StreamHandler(sys.stdout)
+_formatter = jsonlogger.JsonFormatter()
+_logHandler.setFormatter(_formatter)
+_logger.addHandler(_logHandler)
+
+
+class Log:
+    def run_rosettafold_request(self, job_id: str, fasta: bytes | None, a3m: bytes | None):
```
Comment (timurishmuratov7) [inline microservices/reinvent/reinvent/loggers.py:17]: Rosetta fold here instead of reinvent)

File: rosetta_tmp.fasta
Comment (timurishmuratov7) [inline rosetta_tmp.fasta:1]: Is this a file we need?

File: microservices/reinvent/build/Dockerfile [hunk 4]
```diff
@@ -1,33 +1,53 @@
...
-RUN mkdir -p ~/miniconda3
-RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
-RUN bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
-RUN rm -rf ~/miniconda3/miniconda.sh
-ENV PATH="/root/miniconda3/bin:${PATH}"
-ARG PATH="/root/miniconda3/bin:${PATH}"
-RUN conda env create -f reinvent/DockStream/environment.yml
+#FROM nvcr.io/nvidia/cuda:11.3.0-cudnn8-runtime-ubuntu20.04
+FROM nvidia/cuda:12.3.2-runtime-ubuntu22.04
+ARG ROSETTACOMMONS_CONDA_USERNAME
```
Comment (timurishmuratov7) [inline microservices/reinvent/build/Dockerfile:3]: Do we need rosetta credentials here?

File: tmp.fasta
Comment (timurishmuratov7) [inline tmp.fasta:1]: What's this file btw?

**Review by timurishmuratov7 — CHANGES_REQUESTED** [review]

File: microservices/reinvent/reinvent/api_models.py (hunk 2 above)
Comment (JaktensTid) [reply microservices/reinvent/reinvent/api_models.py:12]: It is passed as a parameter to the API method, so it will be able to consume file as bytes

**Review by timurishmuratov7 — APPROVED** [review]

**Pull request merged**